Minimum Spanning Tree/MinimumSpanningTree.playground/Sources/Graph.swift

```diff
@@ -0,0 +1,55 @@
+
+// Undirected edge
+public struct Edge<T>: CustomStringConvertible {
+  public let vertex1: T
+  public let vertex2: T
+  public let weight: Int
+
+  public var description: String {
+    return "[\(vertex1)-\(vertex2), \(weight)]"
+  }
+}
+
+
+// Undirected weighted graph
+public struct Graph<T: Hashable>: CustomStringConvertible {
+
+  public private(set) var edgeList: [Edge<T>]
+  public private(set) var vertices: Set<T>
+  public private(set) var adjList: [T: [(vertex: T, weight: Int)]]
+
+  public init() {
+    edgeList = [Edge<T>]()
+    vertices = Set<T>()
+    adjList = [T: [(vertex: T, weight: Int)]]()
+  }
+
+  public var description: String {
+    var description = ""
+    for edge in edgeList {
+      description += edge.description + "\n"
+    }
+    return description
+  }
+
+  public mutating func addEdge(vertex1 v1: T, vertex2 v2: T, weight w: Int) {
+    edgeList.append(Edge(vertex1: v1, vertex2: v2, weight: w))
+    vertices.insert(v1)
+    vertices.insert(v2)
+    if adjList[v1] == nil {
+      adjList[v1] = [(vertex: v2, weight: w)]
+    } else {
+      adjList[v1]!.append((vertex: v2, weight: w))
+    }
+
+    if adjList[v2] == nil {
```

Comment:
Thanks for pointing it out! I felt a bit uncomfortable about this when I coded this as well.
There's a problem with if let
![screen shot 2017-04-25 at 3 40 59 pm](https://cloud.githubusercontent.com/assets/1306135/25373816/a62662e6-29cd-11e7-89ef-11e656edefca.png)
I was thinking of something like 
```
adjList[v1] = adjList[v1] ?? []()
adjList[v1]?.append((vertex: v2, weight: w))]
```
as well but it's not as efficient since value type assignment does copy the entire value.

I removed the force unwrapping and made it a bit more Swift-ier.